diff --git a/source/dub/dependency.d b/source/dub/dependency.d
index 8817d5e1..22383b4e 100644
--- a/source/dub/dependency.d
+++ b/source/dub/dependency.d
@@ -176,8 +176,8 @@ struct Dependency {
 	/// Returns the exact version matched by the version range.
 	@property Version version_() const @safe {
 		auto range = this.m_value.match!(
-			(NativePath   p) => assert(0),
-			(Repository   r) => assert(0),
+			(NativePath   p) => VersionRange(),
+			(Repository   r) => VersionRange(),
 			(VersionRange v) => v,
 		);
 		enforce(range.isExactVersion(),
